src/main/java/net/minecraftforge/event/terraingen/DecorateBiomeEvent.java

```diff
@@ -49,12 +49,22 @@
     private final World world;
     private final Random rand;
     private final BlockPos pos;
+    private final BlockPos chunkPos;
 
+    public DecorateBiomeEvent(World world, Random rand, BlockPos pos, BlockPos chunkPos)
+    {
+        this.world = world;
+        this.rand = rand;
+        this.pos = pos;
+        this.chunkPos = chunkPos;
+    }
+    
     public DecorateBiomeEvent(World world, Random rand, BlockPos pos)
     {
         this.world = world;
         this.rand = rand;
         this.pos = pos;
+        chunkPos = new BlockPos((pos.getX()/16)*16, (pos.getY()/16)*16, (pos.getZ()/16)*16);
```

Comment:
This calculation is not accurate, `pos` is a random position that can be anywhere in 4 different chunks, so there is no way to know the true origin chunk pos here.
I think it would be better to leave `chunkPos` null in this case and make it and the getter `@Nullable`.
In 1.13 we can remove this old constructor.